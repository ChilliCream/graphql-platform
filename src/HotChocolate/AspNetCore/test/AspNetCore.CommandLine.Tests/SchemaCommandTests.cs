using System.CommandLine.IO;
using System.CommandLine.Parsing;
using CookieCrumble;
using Microsoft.Extensions.Hosting;
using Moq;

namespace HotChocolate.AspNetCore.CommandLine;

public class SchemaCommandTests
{
    [Fact]
    public async Task App_Should_OutputCorrectHelpTest_When_HelpIsRequested()
    {
        // arrange
        var host = new Mock<IHost>().Object;
        var console = new TestConsole();
        var app = new App(host).Build();

        // act
        await app.InvokeAsync("schema -h", console);

        // assert
        console.Out.ToString().MatchSnapshot();
    }
}
