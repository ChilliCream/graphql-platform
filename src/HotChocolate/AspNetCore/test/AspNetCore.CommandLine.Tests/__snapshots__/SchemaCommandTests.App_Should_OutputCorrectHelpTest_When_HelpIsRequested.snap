Description:
  Schema management commands.

Usage:
  graphql schema [command] [options]

Options:
  -?, -h, --help  Show help and usage information

Commands:
  export  Export the graphql schema. If no output (--output) is specified the schema will be printed to the console.

